core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/UserUniqueIDManger.java

```diff
@@ -74,11 +74,27 @@ public User addUser(String username, String profileName, AbstractUserStoreManage
     public User getUser(String uniqueId, AbstractUserStoreManager userStoreManager)
             throws UserStoreException {
 
+        return getUser(uniqueId, userStoreManager, null);
+    }
+
+    /**
+     * Get user from unique id with user store domain present.
+     * @param uniqueId User's unique id.
+     * @param userStoreManager User store manager instance.
+     * @param userStoreDomain User store domain of the user.
+     * @return User object if user presents for the unique id. Null otherwise.
+     */
+    public User getUser(String uniqueId, AbstractUserStoreManager userStoreManager, String userStoreDomain)
+            throws UserStoreException {
+
         String userName = getFromUserNameCache(uniqueId);
         if (StringUtils.isEmpty(userName)) {
-
-            String[] usernames = userStoreManager.getUserList(UserCoreClaimConstants.USER_ID_CLAIM_URI, uniqueId,
-                    null);
+            String uniqueIdWithDomain = uniqueId;
```

Comment:
There is a chance that this variable will contain the unique id without domain. (Based on the below "if" case). So better to avoid this variable and restructure the code to call the method directly with domain appended inside the below "if" case.

Reply:
Had a discussion with @ashensw regarding this earlier. If the domain name is appended to `uniqueId` at this point, then it needs to be removed again before the caching methods at line 106.

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/3417/commits/6bb7a2001f36fb521025886bcf73b1738cf286ab